catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/transformer/TransformerDescriptors.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.transformer;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.osgi.framework.ServiceReference;
+
+public class TransformerDescriptors {
+
+  private List<Map<String, String>> metacardTransformers;
+
+  private List<Map<String, String>> queryResponseTransformers;
+
+  public TransformerDescriptors(
+      List<ServiceReference> metacardTransformers,
+      List<ServiceReference> queryResponseTransformers) {
+    this.metacardTransformers = getTransformerDescriptors(metacardTransformers);
```

Comment:
This is better but it will still not handle transformers being added or removed at runtime. These lists will be set with the currently available transformers at the time this object is created and will not change after that. I was thinking that this object would maintain two lists of `ServiceReference`s instead of maps and iterate over them to get the transformer properties. Then `getMetacardTransformers()` and `getQueryResponseTransformers()` could look like:
```java
public List<Map<String, String>> getMetacardTransformers() {
  return getTransformerDescriptors(metacardTransformers);
}

public List<Map<String, String>> getQueryResponseTransformers() {
  return getTransformerDescriptors(queryResponseTransformers);
}
```
where `metacardTransformers` and `queryResponseTransformers` are `List<ServiceReference>`s.

Then you would just need to add `Map<String, String> getMetacardTransformer(String id)` and `Map<String, String> getQueryResponseTransformer(String id)` functions to make things easier for clients of this class.